kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/EmbeddedKafkaIntegrationTest.java

```diff
@@ -1,39 +1,49 @@
 package uk.gov.homeoffice.digital.sas.kafka;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
 
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.kafka.test.context.EmbeddedKafka;
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.TestPropertySource;
 import uk.gov.homeoffice.digital.sas.config.TestConfig;
-import uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumer;
+import uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumerServiceImpl;
 import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaEventMessage;
 import uk.gov.homeoffice.digital.sas.kafka.producer.KafkaProducerService;
 import uk.gov.homeoffice.digital.sas.model.Profile;
 import uk.gov.homeoffice.digital.sas.repository.ProfileRepository;
 
 @SpringBootTest(classes = TestConfig.class)
 @DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
 @EmbeddedKafka(
-    partitions = 1,
+    partitions = 3,
     brokerProperties = {
-        "listeners=PLAINTEXT://localhost:3333",
-        "port=3333"
+        "listeners=PLAINTEXT://localhost:2111",
```

Comment:
Just wondering about some of these settings changing. I think by default it is 0 which means an ephemeral port is allocated